pulp/tests/test_pulp.py

```diff
@@ -1103,16 +1104,42 @@ def test_infeasible_problem__is_not_valid(self):
         prob += 1 * x
         prob += x >= 2  # Constraint x to be more than 2
         prob += x <= 1  # Constraint x to be less than 1
-        if self.solver.name in ['GUROBI_CMD']:
+        if self.solver.name in ["GUROBI_CMD"]:
             pulpTestCheck(
-                prob, self.solver, [const.LpStatusNotSolved, const.LpStatusInfeasible, const.LpStatusUndefined]
+                prob,
+                self.solver,
+                [
+                    const.LpStatusNotSolved,
+                    const.LpStatusInfeasible,
+                    const.LpStatusUndefined,
+                ],
             )
         else:
             pulpTestCheck(
                 prob, self.solver, [const.LpStatusInfeasible, const.LpStatusUndefined]
             )
         self.assertFalse(prob.valid())
 
+    def test_measuring_solving_time(self):
+        print("\t Testing measuring optimization time")
+
+        time_limit = 10.23
+        solver_settings = dict(PULP_CBC_CMD=30, COIN_CMD=30, SCIP_CMD=30)
+
+        bins = solver_settings.get(self.solver.name)
+        if bins is not None:
+            prob = create_bin_packing_problem(bins=bins)
+            self.solver.timeLimit = time_limit
+            prob.solve(self.solver)
+
+            self.assertAlmostEqual(
+                prob.solutionTime,
+                time_limit,
+                delta=2,
```

Comment:
Unfortunately, CBC shows higher variations in the measured time, so smaller `delta`s failed.
Could be tested with solver specific settings.